### PR TITLE
Page creation and duplication: Decode HTML entities in success notices

### PR DESCRIPTION
**Pull request by carolinan**
## What?

Decodes HTML entities in the post and page titles in the success notices that shows when creating a new page through the Site Editor, and when duplicating a post or page via the Site Editor Pages screen or the block editor.

**Note: This meant adding "@wordpress/html-entities" to the edit-post package.**

Partial for https://github.com/WordPress/gutenberg/issues/60995
Replaces #61086

## Why?

When a page or post title has HTML entities like & in the post title, these success notices displayed the characters as encoded. So & displayed as `&#038;`.

## How?

Wrap the titles inside `decodeEntities` https://developer.wordpress.org/block-editor/reference-guides/packages/packages-html-entities/#decodeentities

## Testing Instructions
- Go to Appearance > Editor > Pages.
- Create a new Page, add a & to the title, or some other HTML entity you want to test.
- Because your new page is blank, the pattern selection modal may pop up: you may need to close it to see the snackbar notice in the bottom corner.
- Confirm that all characters in the notice display correctly, and are not encoded.
- Locate the Actions menu in the Page panel.
- Select "Duplicate" in the drop down.
- Do not change the title, keep the character you added in the previous step. Duplicate.
- Confirm that all characters in the notice display correctly, and are not encoded.

### CODE REVIEW

File: packages/edit-site/src/components/editor/index.js [hunk 1]
```diff
@@ -176,7 +177,7 @@ export default function Editor( { isLoading } ) {
 							sprintf(
 								// translators: %s: Title of the created post e.g: "Post 1".
 								__( '"%s" successfully created.' ),
-								_title
+								decodeEntities( _title )
```
Comment (carolinan) [inline packages/edit-site/src/components/editor/index.js:150]: @jorgefilipecosta I am not sure where in the user interface this is visible and how to test it?

Comment (carolinan) [reply packages/edit-site/src/components/editor/index.js:150]: I'm pretty sure I have asked this before but already forgotten.😞

Comment (Mamaduka) [reply packages/edit-site/src/components/editor/index.js:150]: I think the notice is for post duplication, though the action might have been disabled for the moment.

**Review by Mamaduka — APPROVED** [review]
Looks good ✅

File: packages/edit-site/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 ## Unreleased
 
+### Enhancements
+
+-   Add ´@wordpress/html-entities´ package to the list of dependencies in package.json. ([#62313](https://github.com/WordPress/gutenberg/pull/62313))
```
Comment (ellatrix) [inline packages/edit-site/CHANGELOG.md]: This PR is filed as a bug fix, but here it's filed an enhancement.

**Comment by ellatrix** [comment]
Shall we backport this for WP 6.6? Please add the `Backport to WP Beta/RC` label if it’s a good candidate for backporting. Thanks!

File: packages/edit-site/CHANGELOG.md (hunk 2 above)
Comment (Mamaduka) [reply packages/edit-site/CHANGELOG.md]: This is also the wrong package change log. The package was added to the `edit-post`.

**Comment by Mamaduka** [comment]
I added the backport label since the issue doesn't exist in WP 6.5.

File: packages/edit-site/CHANGELOG.md (hunk 2 above)
Comment (carolinan) [reply packages/edit-site/CHANGELOG.md]: Thank you, what a lazy mistake from my side!

**Comment by ellatrix** [comment]
I manually cherry picked for 6.6 this because there were conflicts.

**Pull request merged**